### PR TITLE
fix(api,worker): only delete terminated jobs + fix cache with symlinks

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12091.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

File: engine/api/workflow/dao_node_run_job.go [hunk 1]
```diff
@@ -262,8 +262,8 @@ func LoadNodeJobRun(ctx context.Context, db gorp.SqlExecutor, store cache.Store,
 //LoadDeadNodeJobRun load a NodeJobRun which is Building but without worker
```
Comment (richardlt) [inline engine/api/workflow/dao_node_run_job.go]:
```suggestion
//LoadDeadNodeJobRun load all NodeJobRun without worker
```

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12091.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12091.2 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Pull request merged**